crates/revm/src/context/context_precompiles.rs

```diff
@@ -144,29 +130,28 @@ impl<DB: Database> ContextPrecompiles<DB> {
     /// Clones the precompiles map if it is shared.
     #[inline]
     pub fn to_mut(&mut self) -> &mut HashMap<Address, ContextPrecompile<DB>> {
-        match self.inner {
-            PrecompilesCow::Default(_) => self.to_mut_slow(),
-            PrecompilesCow::Owned(ref mut inner) => inner,
-        }
-    }
+        self.mutate_into_owned();
 
-    #[allow(clippy::wrong_self_convention)]
-    #[cold]
-    fn to_mut_slow(&mut self) -> &mut HashMap<Address, ContextPrecompile<DB>> {
```

Comment:
I don't understand this change

Reply:
It is refactor of code so it reads better, the functionality is same

Reply:
to_mut_slow is marked as cold because it's only done once, but to_mut is called in every other API, pls keep that, it's intentional

Reply:
Cold does not do a lot here, that is why it is removed. `to_mut` is called only once (or few times) per transaction.

`mutate_into_owned` is going to be inlined and the current overhead is one pattern match (maybe it is even optimized), i dont see cold making any significant difference.

Reply:
Makes the fast path the one that's inlined, and the slow path that is called only once not inlined. Because this method is used in multiple places it makes sense, see `Cow::to_mut`

Reply:
Is this significant at all? That is my question, there is not a lot of sense in optimizing every `if` statement.

Reply:
It is used only to extend precompile list

Reply:
Sure, but if it was already done by me, why remove it? :D

Reply:
Dont you think that? Optimisations are worth only on hot paths. Borrowed -> Owned path is slow, but check if it `Borrowed` is not, and besides, branch prediction does good job. The code now looks better and is nicer to read, in this case I optimised for that.

Reply:
No because it's not about speed but code size, you are inlining both the check and hot path at every `to_mut` callsite. This is the same situation as `OnceLock` for example, where 99% of the time it's called it will return the initialized value, otherwise it calls the initializer and returns the value from there

Reply:
That is why i didn't understand you: "not about speed but code size.". `OnceLock` is a good example.

It is not going to matter a lot as `to_mut` is used only in extending precompiles, but it is a cool mechanism to have, so I will change it to that.

Reply:
reintroduced cold function https://github.com/bluealloy/revm/pull/1486/commits/fa38da59a94ce43a8b508e0342a860a3ed81fa39 
@DaniPopes can you take a look